pkg/api/handlers/compat/images_search.go

```diff
@@ -34,13 +34,19 @@ func SearchImages(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	_, authfile, err := auth.GetCredentials(r)
+	authconf, authfile, err := auth.GetCredentials(r)
 	if err != nil {
 		utils.Error(w, http.StatusBadRequest, err)
 		return
 	}
 	defer auth.RemoveAuthfile(authfile)
 
+	var username, password string
+	if authconf != nil {
+		username = authconf.Username
+		password = authconf.Password
```

Comment:
I think this should also read `IdentityToken`, and pass it to c/common.

(I don’t think it’s necessary to expose it in the CLI or in the API client at Podman level, but if the data can exist at this code path, knowingly discarding it would not save much time now, and could cost us debugging effort in the future.)